WalletWasabi/Tor/Http/TorHttpClient.cs

```diff
@@ -178,25 +171,29 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,Canc
 			request.Version = HttpProtocol.HTTP11.Version;
 			request.Headers.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
 
-			if (TorSocks5Client is { } && !TorSocks5Client.IsConnected)
+			if (TorSocks5Client is null)
```

Comment:
I would not check for the property, it is a bad practice because the property can be not null in a time `t` and be null in `t+1` so, checking makes no sense. Instead I would use a local variable and also I would extract the code segment to a new method. Something like this:

```c#
var torSocks5Client = TorSocks5Client ?? CreateTorSocks5Client();
```

And then use the local variable in all future references.

Reply:
It mostly works because `TorSocks5Client` is used under `using (await AsyncLock.LockAsync(cancel).ConfigureAwait(false))` lock held in `SendAsync` (with relative URI). 

However, there are two cases when it is not so and that's a bug.

Reply:
These two bugs are fixed in my #4738. I would like to fix that in some follow-up PR. The bugs are present now anyway.

Reply:
ok